test/validation/api/ipsec/ipsec_test_out.c

```diff
@@ -1591,6 +1591,150 @@ static void test_test_sa_update_seq_num(void)
+	uint64_t hard_limit_byte;
+	uint64_t soft_limit_pkt;
+	uint64_t hard_limit_pkt;
+	odp_ipsec_sa_t out_sa;
+
+	switch (expiry)	{
+	case IPSEC_TEST_EXPIRY_SOFT_PKT:
+		soft_limit_pkt = SOFT_LIMIT_PKT_CNT;
+		hard_limit_pkt = HARD_LIMIT_PKT_CNT;
+		soft_limit_byte = 0;
+		hard_limit_byte = 0;
+		delta = DELTA_PKT_CNT;
+		limit = soft_limit_pkt;
+		inc = 1;
+		break;
+	case IPSEC_TEST_EXPIRY_HARD_PKT:
+		soft_limit_pkt = SOFT_LIMIT_PKT_CNT;
+		hard_limit_pkt = HARD_LIMIT_PKT_CNT;
+		soft_limit_byte = 0;
+		hard_limit_byte = 0;
+		delta = DELTA_PKT_CNT;
+		limit = hard_limit_pkt;
+		inc = 1;
+		break;
+	case IPSEC_TEST_EXPIRY_SOFT_BYTE:
+		soft_limit_pkt = 0;
+		hard_limit_pkt = 0;
+		soft_limit_byte = test_packet_len * SOFT_LIMIT_PKT_CNT;
+		hard_limit_byte = test_packet_len * HARD_LIMIT_PKT_CNT;
+		delta = test_packet_len * DELTA_PKT_CNT;
+		limit = soft_limit_byte;
+		inc = test_packet_len;
+		break;
+	case IPSEC_TEST_EXPIRY_HARD_BYTE:
+		soft_limit_pkt = 0;
+		hard_limit_pkt = 0;
+		soft_limit_byte = test_packet_len * SOFT_LIMIT_PKT_CNT;
+		hard_limit_byte = test_packet_len * HARD_LIMIT_PKT_CNT;
+		delta = test_packet_len * DELTA_PKT_CNT;
+		limit = hard_limit_byte;
+		inc = test_packet_len;
+		break;
+	default:
+		return;
+	}
+
+	memset(&out_tunnel, 0, sizeof(odp_ipsec_tunnel_param_t));
+
+	out_tunnel.type = ODP_IPSEC_TUNNEL_IPV4;
+	out_tunnel.ipv4.src_addr = &src;
+	out_tunnel.ipv4.dst_addr = &dst;
+
+	CU_ASSERT(odp_ipsec_capability(&capa) == 0);
+
+	ipsec_sa_param_fill(&param_out, ODP_IPSEC_DIR_OUTBOUND, ODP_IPSEC_ESP,
+			    0x4a2cbfe7, &out_tunnel,
+			    ODP_CIPHER_ALG_AES_CBC, &key_a5_128,
+			    ODP_AUTH_ALG_SHA1_HMAC, &key_5a_160,
+			    NULL, NULL);
+
+	param_out.lifetime.soft_limit.bytes = soft_limit_byte;
+	param_out.lifetime.hard_limit.bytes = hard_limit_byte;
+	param_out.lifetime.soft_limit.packets = soft_limit_pkt;
+	param_out.lifetime.hard_limit.packets = hard_limit_pkt;
+
+	out_sa = odp_ipsec_sa_create(&param_out);
+	CU_ASSERT_NOT_EQUAL_FATAL(ODP_IPSEC_SA_INVALID, out_sa);
+
+	ipsec_test_part test_out = {
+		.pkt_in = &pkt_ipv4_icmp_0,
+		.num_pkt = 1,
+		.out = {
+			{ .status.warn.all = 0,
+			  .status.error.all = 0,
+			  .l3_type = ODP_PROTO_L3_TYPE_IPV4,
+			  .l4_type = ODP_PROTO_L4_TYPE_ESP,
+			},
+		},
+	};
+
+	if ((suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_INLINE) &&
+	    (expiry == IPSEC_TEST_EXPIRY_HARD_PKT ||
+	     expiry == IPSEC_TEST_EXPIRY_HARD_BYTE)) {
+		test_out.flags.test_sa_expiry = true;
+		test_out.out[0].sa_expiry = IPSEC_TEST_EXPIRY_NONE;
+	}
+
+	for (i = 0; i < limit - delta; i += inc)
+		ipsec_check_out_one(&test_out, out_sa);
+
+	sa_expiry_notified = false;
+	test_out.flags.test_sa_expiry = true;
+	test_out.out[0].sa_expiry = expiry;
+
+	for (; i <= limit; i += inc) {
+		ipsec_check_out_one(&test_out, out_sa);
+		if (sa_expiry_notified)
+			break;
+	}
+
+	if (suite_context.outbound_op_mode == ODP_IPSEC_OP_MODE_INLINE)
+		ipsec_status_event_get(out_sa, expiry, true);
```

Comment:
Why is this needed? For handling status events that ipsec_check_out_one() failed to handle? Maybe ipsec_check_out_one() should then be improved instead?

Reply:
Ack. For soft expiry status event test, with 0 wait time for receiving IPsec status event, observation was that the status event was received a little after the entire test packets were run. By having a 1 msec wait time for each IPsec status event receive calls, the test seems to run fine. So removed this handling with v9.